homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -343,23 +356,29 @@ async def _async_replace_device(self, replace_device):
 
         device_registry.async_remove_device(old_device)
 
-    def _can_add_device(self, new_rfx_obj):
+    def _can_add_device(self, new_rfx_obj: rfxtrxmod.RFXtrxEvent) -> bool:
         """Check if device does not already exist."""
         new_device_id = get_device_id(new_rfx_obj.device)
         for packet_id, entity_info in self._config_entry.data[CONF_DEVICES].items():
             rfx_obj = get_rfx_object(packet_id)
+            assert rfx_obj
+
             device_id = get_device_id(rfx_obj.device, entity_info.get(CONF_DATA_BITS))
             if new_device_id == device_id:
                 return False
 
         return True
 
-    def _can_replace_device(self, entry_id):
+    def _can_replace_device(self, entry_id: str) -> bool:
```

Comment:
We could pass the  `_can_replace_device` attribute as a parameter here (and make this function a standalone function). Then we wouldn't need to assert below.

Reply:
Sure, but i'd rather not change logic in a pull to add full typing.

Reply:
While it's typically a good idea to have small focused PRs that aren't doing too much, typing is an interesting case where it points out issues with the existing code made more obvious when you have to jump through hoops with respect to the typing. I think we sometimes bend when the refactoring is related to the typing in that way.

Alternatively, you can send a "pre-factor" PR that gets merged before this one that changes the code, then have this PR just about typing?

Reply:
I've approved the PR. Refactoring can be done later.